omod/src/main/webapp/pages/metadata/roles/manageRoles.gsp

```diff
@@ -1,67 +1,55 @@
 <%
     ui.decorateWith("appui", "standardEmrPage")
-    ui.includeCss("adminui", "account.css")
+
+    ui.includeJavascript("uicommons", "angular.min.js")
+    ui.includeJavascript("uicommons", "angular-resource.min.js")
+    ui.includeJavascript("uicommons", "angular-ui/angular-ui-router.min.js")
+    ui.includeJavascript("uicommons", "angular-app.js")
+    ui.includeJavascript("uicommons", "angular-common.js")
+    ui.includeJavascript("uicommons", "services/roleService.js")
+    ui.includeJavascript("uicommons", "services/privilegeService.js")
+    ui.includeJavascript("uicommons", "filters/display.js")
+    ui.includeJavascript("uicommons", "filters/serverDate.js")
+
+    ui.includeJavascript("uicommons", "ngDialog/ngDialog.js")
+    ui.includeCss("uicommons", "ngDialog/ngDialog.min.css")
+
+    ui.includeJavascript("adminui", "metadata/colTable.js")
+    ui.includeJavascript("adminui", "metadata/manageRoles.js")
 %>
+
 <script type="text/javascript">
     var breadcrumbs = [
-		{ icon: "icon-home", link: '/' + OPENMRS_CONTEXT_PATH + '/index.htm' },
-		{ label: "${ ui.message('adminui.app.configureMetadata.label')}" , link: '${ui.pageLink("adminui", "metadata/configureMetadata")}'},
-		{ label: "${ ui.message("adminui.manageRoles.title")}" }
+        { icon: "icon-home", link: '/' + OPENMRS_CONTEXT_PATH + '/index.htm' },
+        { label: "${ ui.message('adminui.app.configureMetadata.label')}", link: '${ui.pageLink("adminui", "metadata/configureMetadata")}'},
+        { label: "${ ui.message("adminui.manageRoles.title")}" }
     ];
+    emr.loadMessages([
+    	"adminui.role.purge.success",
+    	"adminui.role.purge.error",
+    	"adminui.role.getRoles.error",
+    	"adminui.role.getPrivileges.error",
+    	"adminui.role.save.success",
+    	"adminui.role.save.error",
+        "uicommons.generalSavedNotification"
+    ]);
 </script>
 
-<% i=0 %>
-
-<h3>${ ui.message("adminui.manageRoles.title") }</h3>
-
-<input type="submit" class="button" value="${ui.message("adminui.addNewRole.label")}" onclick="window.location='/${ contextPath }/adminui/metadata/roles/role.page'"/>
+<style type="text/css">
```

Comment:
Actually this is old code not present in the latest changes. In the new angular code, the add button is on the list.gsp page.